### PR TITLE
Add ability to dynamicaly skip a prompt

### PR DESCRIPTION
**Pull request by fujifish**
It is sometimes useful to control whether to display a prompt to the user based on previous answers.
This PR adds a new `ask` method to a property that enables to skip the prompt based on custom logic.

E.g.:
``` js
  var schema = {
    properties: {
      proxy: {
        description: 'Proxy url',
      },
      proxyCredentials: {
        description: 'Proxy credentials',
        ask: function() {
          // only ask for proxy credentials if a proxy was set
          return prompt.history('proxy').value > 0;
        }
      }
    }
  };
```

I've also included two other minor enhancements:
1. If `prompt.message` is set to an empty string then do not print the delimiter
2. If a property is not valid and has a `message` defined, then do not print the generic "Invalid input for ..." message.

Great project by the way!

### CODE REVIEW

File: lib/prompt.js [hunk 1]
```diff
@@ -416,9 +416,11 @@ prompt.getInput = function (prop, callback) {
...
   defaultLine = schema.default;
   name = prop.description || schema.description || propName;
-  raw = prompt.colors
-    ? [prompt.message, delim + name.grey, delim.grey]
-    : [prompt.message, delim + name, delim];
+  // only show the message delimiter if there is actually a message
+  var prefix = prompt.message && prompt.message.length > 0 ? prompt.message + delim : '',
+    raw = prompt.colors
+      ? [prefix, name.grey, delim.grey]
+      : [prefix, name, delim];
```
Comment (gangstead) [inline lib/prompt.js]: This functionality has been fixed in the created-before-this-but-only-recently-merged #61

File: README.md [hunk 2]
```diff
@@ -199,6 +199,45 @@ prompting the user.
...
+  prompt.get(schema, function (err, result) {
+    //
+    // Log the results.
+    //
+    console.log('Command-line input received:');
+    console.log('  proxy: ' + result.proxy);
+    console.log('  credentials: ' + result.proxyCredentials);
+  });
+```
+
```
Comment (gangstead) [inline README.md]: Can you also add this to the examples folder?

**Comment by gangstead** [comment]
Sorry this sat for so long, I've started helping on this project (yesterday).  This looks really good, thanks for including updates to the docs too.  

I've put some other changes in which makes this need an update. Do you think you could do a pull and fix the conflicts so I can accept this PR?

**Comment by fujifish** [comment]
done both requests.

**Comment by gangstead** [comment]
I fixed the test that was failing.  It was looking for "Invalid Input" but now that extra line doesn't show up if schema.message is set.

**Comment by gangstead** [comment]
Thanks!

**Comment by fujifish** [comment]
Thanks. When is a release to NPM expected?

**Comment by gangstead** [comment]
As son as i can get permission i will. Working on it with the admins #144

**Comment by jcrugzz** [comment]
`v0.3.0` is relased in `npm`

**Pull request merged**